### ISSUE TITLE
Need more strict validation on specific value in YAML file

### ISSUE THREAD

**Issue by songkaiape** [Azure/vscode-kubernetes-tools#146]
For example: 
The ports value should be an integer between 0 ~65536. But plugin only validate if the value is integer. Invalid value like 888888 or -1 should not pass validation.

ImagePullPolicy should only be one of Always, Never, IfNotPresent. Now it only validate if it is a string

**Comment by songkaiape** [comment]
screenshot
![image](https://user-images.githubusercontent.com/4318793/38072397-a5be5256-3358-11e8-954b-a5ffaa27112f.png)

**Comment by bacongobbler** [comment]
Helm 3 introduces JSONSchema support for chart values (see https://github.com/helm/helm/pull/5350). Once Helm 3 lands, the extension could load the `values.schema.json` (if present) to provide validation against what the user has provided in the values.yaml. That should handle use cases such as the few cases provided here.

**Comment by thernstig** [comment]
@bacongobbler I just tried to add a `values.schema.json`. I also added this:

```json
  "yaml.schemas": {
    "charts/server/values.schema.json": [
      "charts/server/values.yaml"
    ]
  }
```
and noticed that when I tried to auto-comlete things in in values.yaml in VS Code, it did not give me IntelliSense for the items I had specified. Since Helm 3 has been out for quite some time now, (and Helm 2 has been deprecated), is there a chance this feature could get some priority? :)

I have no idea how you relate this to the original issue in this thread though, so if you want me to I can make this a separate issue. The original issue by @songkaiape is about a Deployment manifest. That has nothing to do with `values.schema.json` which should only be about validating the `values.yaml` file.

**Comment by thernstig** [comment]
In addition, https://github.com/redhat-developer/vscode-yaml/issues/196 seems to have an issue here as well. What is the reason the `values.yaml` file has this identifier? I need to manually change the language mode to `yaml` for the `yaml.schemas` setting to work with my `values.yaml`.

```sh
"languages": [
            {
                "id": "helm",
                "aliases": [
                    "helm-template",
                    "helm"
                ],
                "filenamePatterns": [
                    "values.yaml",
                    "Chart.yaml",
                    ......
```
